File: typescript-client-sdk/src/sdk/models/operations/getdatacompaniescompanyidassessdatatypesdatatypedataintegritydetails.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=dataType" })
  dataType: string;
}


export class GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=orderBy" })
  orderBy?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=query" })
  query?: string;
}


export class GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsQueryParams;

  @SpeakeasyMetadata()
  security: GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsSecurity;
}


export class GetDataCompaniesCompanyIdAssessDataTypesDataTypeDataIntegrityDetailsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataIntegrityContractsDetailsTransactionDetailsPagedResponse?: shared.CodatDataIntegrityContractsDetailsTransactionDetailsPagedResponse;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getdatacompaniescompanyidassessdatatypesdatatypedataintegritydetails.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
	DataType  string `pathParam:"style=simple,explode=false,name=dataType"`
}

type GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsQueryParams struct {
	OrderBy  *string `queryParam:"style=form,explode=true,name=orderBy"`
	Page     *int    `queryParam:"style=form,explode=true,name=page"`
	PageSize *int    `queryParam:"style=form,explode=true,name=pageSize"`
	Query    *string `queryParam:"style=form,explode=true,name=query"`
}

type GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsRequest struct {
	PathParams  GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsPathParams
	QueryParams GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsQueryParams
	Security    GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsSecurity
}

type GetDataCompaniesCompanyIDAssessDataTypesDataTypeDataIntegrityDetailsResponse struct {
	CodatDataIntegrityContractsDetailsTransactionDetailsPagedResponse *shared.CodatDataIntegrityContractsDetailsTransactionDetailsPagedResponse
	ContentType                                                       string
	StatusCode                                                        int
}
